thirdeye/thirdeye-frontend/app/pods/sandbox/controller.js

```diff
@@ -0,0 +1,136 @@
+import Ember from 'ember';
+import moment from 'moment';
+import fetch from 'fetch';
+
+const generateRandomData = (num, min=0, max=100) => {
+  const diff = max - min;
+  const dataPoints = [];
+  for (let i = 0; i < num; i++) {
+    dataPoints.push(Math.random() * diff + min);
+  }
+  return dataPoints;
+};
+
+const generateSequentialData = (num, min=0, max=100) => {
+  const diff = max - min;
+  const dataPoints = [];
+  for (let i = 0; i < num; i++) {
+    dataPoints.push(min + diff * (i / num));
+  }
+  return dataPoints;
+};
+
+export default Ember.Controller.extend({
+
+  types: null,
+
+  colors: null,
+
+  axes: null,
+
+  series: Ember.computed(
+    'model.data',
+    'types',
+    'colors',
+    'axes',
+    function() {
+      console.log('series()');
+
+      const types = this.get('types') || {};
```

Comment:
if you're checking against `undefined` you can use `this.getWithDefault('prop', 'defaultValueIfUndefined');`
caveat: this does not work when `null`

Reply:
deleted